tensornetwork/contractors/stochastic_contractor.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2019 The TensorNetwork Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Stochastic Network Contraction."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import random
+import numpy as np
+from tensornetwork import tensornetwork
+
+
+def get_random_edge(network: tensornetwork.TensorNetwork) -> tensornetwork.Edge:
+  """Returns a random connected edge.
+  If it finds a trace edge it contracts it.
+  Args:
+    network: Connected TensorNetwork we are trying to contract.
+  Returns:
+    edge: A random connected edge from the network.
+  """
+  node = random.choice(tuple(network.nodes_set))
+  edge = random.choice(node.edges)
```

Comment:
I believe we have a node.get_nondanglinge_edges() method now. You can use that instead of this while loop (which will break if you have a node with only dangling edges.